src/Driver.cpp

```diff
@@ -178,9 +142,208 @@ void Driver::startAcquisition()
+                                    bool automaticPingCycling,
+                                    bool binaryDataOutput,
+                                    bool enableSerialOutput,
+                                    bool enableDataRecording)
+{
+    setConfigurationMode();
+    writePacket(reinterpret_cast<uint8_t const*>(("CF" + std::to_string(automaticEnsembleCycling)
+                                                      + std::to_string(automaticPingCycling)
+                                                      + std::to_string(binaryDataOutput)
+                                                      + std::to_string(enableSerialOutput)
+                                                      + std::to_string(enableDataRecording)
+                                                      + "\n").c_str()), 8);
+    readConfigurationAck(m_read_timeout);
+}
+
+void Driver::setHeadingAlignment(base::Angle headingAlignment)
+{
+    // Convert from radians to hundredths of a degree.
+    int headingAlignmentInt = static_cast<int>(headingAlignment.getDeg() * 100);
+    sendStandardCommand(std::string("EA"), headingAlignmentInt, 5, true);
+}
+
+void Driver::set_e_headingBias(base::Angle e_headingBias)
+{
+    // Convert from radians to hundredths of a degree.
+    int headingBiasInt = static_cast<int>(e_headingBias.getDeg() * 100);
+    sendStandardCommand(std::string("#EV"), headingBiasInt, 5, true);
+}
+
+void Driver::setSalinity(int salinity)
+{
+    sendStandardCommand(std::string("ES"), salinity, 2);
+}
+
+void Driver::setSensorSourceSettings(SENSORSOURCE speedOfSoundSource,
+                                     SENSORSOURCE depthSource,
+                                     SENSORSOURCE headingSource,
+                                     SENSORSOURCE pitchAndRollSource,
+                                     SENSORSOURCE salinitySource,
+                                     SENSORSOURCE temperatureSource)
+{
+    setConfigurationMode();
+    writePacket(reinterpret_cast<uint8_t const*>(("EZ" + std::to_string(speedOfSoundSource)
+                                                      + std::to_string(depthSource)
+                                                      + std::to_string(headingSource)
+                                                      + std::to_string(pitchAndRollSource)
+                                                      + std::to_string(salinitySource)
+                                                      + std::to_string(temperatureSource)
+                                                      + "0\n").c_str()), 10);
+    readConfigurationAck(m_read_timeout);
+}
+
+void Driver::setTimePerEnsemble(const base::Time& timePerEnsemble)
+{
+    setConfigurationMode();
+    std::vector<int> timeValues = timePerEnsemble.toTimeValues();
+    std::vector<std::string> timeStrings;
+
+    for (int i = 1; i < 5; ++i)
+    {
+        if (i == 1)
+            timeValues.at(i) /= 10;
```

Comment:
Since you want to do this only for one element why doing it inside the loop?

Reply:
Good point. Moved it in front of the loop.